src/core/handle.ml

```diff
@@ -133,6 +134,89 @@ let handle_require_as : popt -> sig_state -> Path.t -> ident -> sig_state =
   let path_map = PathMap.add p id.elt ss.path_map in
   {ss with aliases; path_map}
 
+(** Symbol properties needed for the signature are packed together *)
+type sig_symbol =
+  { expo   : expo        ;
+    prop   : prop        ;
+    mstrat : match_strat ;
+    x      : ident       ;
+    a      : term        ;
+    impl   : bool list   ;
+    d      : term option ; }
+
+(** Wrapper around Sig_state.add_symbol using the sig_symbol type *)
+let add_symbol : sig_state -> sig_symbol -> sig_state =
+  fun ss {expo;prop;mstrat;x;a;impl;d} ->
+  Sig_state.add_symbol ss expo prop mstrat x a impl d
+
+(** [data_proof] returns the datas needed for the symbol or definition
+   [sig_symbol] in the signature and the [goals] we wish to prove with
+   the proof script [ts pe]. [pdata_expo] set the authorized exposition
+   of the symbols used in the proof script : Public (= only public
+   symbols) or Privat (= public and private symbols) *)
+let data_proof sig_symbol cmd pdata_expo ts pe goals =
```

Comment:
please follow the coding style by giving the types first

Reply:
fixed in 7b8c56a